.github/workflows/ckeck-pr-title.yml

```diff
@@ -0,0 +1,20 @@
+name: PR-Title
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+jobs:
+  test:
+    name: Check PR Title
+    runs-on: ubuntu-latest
+    steps:
+      # check PR title
+      - uses: deepakputhraya/action-pr-title@master
+        with:
+          regex: '([a-z])+(\(([a-z\-_])+\))?!?: ([a-z])+' # Regex the title should match.
```

Comment:
Would it make sense to add something like "\(DSP-\d+\)" to the end of the regex?
By convention, the PR's title should end with "(DSP-123)".

It's is actually missing from this PR's title :-)

Reply:
Not all PRs have a task on youtrack. So, this can be tricky. And for example, in another repo I had a task of type `PM-` and not `DSP-`.

Reply:
You're right that it's missing here in the PR title... ;-)  
But I also think it might become at some point problematic to enforce it. (There are also some tasks with `BS-123`. I'm not sure this will be the case in the future, so I think it's a bit risky to enforce this strictly.)

Reply:
maybe make it more general: expect something like: `\(.+-\d+\)`

Reply:
however, I leave this up to you

Reply:
>maybe make it more general: expect something like: `\(.+-\d+\)`

good idea, I'll add that

Reply:
I went even stricter: `\([A-Z]+-\d+\)`

Do you think it's a problem that I allow only lower case letters in the PR title?  Should probably be extended to upper case too,  right? (e.g. for abbreviations that often use upper case.)

Reply:
> I went even stricter: `\([A-Z]+-\d+\)`

you are reading my thoughts :-)

> 
> Do you think it's a problem that I allow only lower case letters in the PR title? Should probably be extended to upper case too, right? (e.g. for abbreviations that often use upper case.)

no idea

Reply:
I'll just add upper case, just to be sure

Reply:
you can never be too paranoid

Reply:
we could argue about that ;-)